x-pack/test/functional/page_objects/security_page.ts

```diff
@@ -17,53 +18,114 @@ export function SecurityPageProvider({ getService, getPageObjects }) {
   const userMenu = getService('userMenu');
   const PageObjects = getPageObjects(['common', 'header', 'settings', 'home', 'error']);
 
-  class LoginPage {
-    async login(username, password, options = {}) {
-      const [superUsername, superPassword] = config.get('servers.elasticsearch.auth').split(':');
+  interface LoginOptions {
+    expectSpaceSelector?: boolean;
+    expectSuccess?: boolean;
+    expectForbidden?: boolean;
+  }
+
+  type LoginExpectedResult = 'spaceSelector' | 'error' | 'chrome';
+
+  async function waitForLoginPage() {
+    await retry.waitForWithTimeout('login page', config.get('timeouts.waitFor') * 5, async () => {
+      const alert = await browser.getAlert();
```

Comment:
Is this alert logic here to cleanup any tests that may have a pending browser confirmation dialog?

Reply:
To be honest, have no idea - just copied existing stuff. But let's try to remove it and see how that goes - we can always return it back (with a proper clarifying comment) if it happens to be necessary.

Reply:
Okay, now we know why we need this code :slightly_smiling_face: 

![alert](https://user-images.githubusercontent.com/1713708/81928084-04df1680-95e5-11ea-8152-cb862e19d0f7.png)